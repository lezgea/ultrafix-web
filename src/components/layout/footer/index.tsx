"use client";

import React from 'react';
import { usePathname } from 'next/navigation';
import dynamic from 'next/dynamic';
import Link from 'next/link';

const InstagramIcon = dynamic(() => import('@assets/icons').then(mod => mod.InstagramIcon), { ssr: false });
const TwitterIcon = dynamic(() => import('@assets/icons').then(mod => mod.TwitterIcon), { ssr: false });
const YoutubeIcon = dynamic(() => import('@assets/icons').then(mod => mod.YoutubeIcon), { ssr: false });
const LinkedinIcon = dynamic(() => import('@assets/icons').then(mod => mod.LinkedinIcon), { ssr: false });
const LogoWhite = dynamic(() => import('@assets/icons').then(mod => mod.LogoWhite), { ssr: false });


export const Footer: React.FC = () => {
    const pathname = usePathname();

    const hideHeaderRoutes = React.useMemo(() => ["/sign-in", "/sign-up", "/activation", "/forgot", "/reset-password"], []);
    const shouldHideFooter = React.useMemo(() => hideHeaderRoutes.includes(pathname), [pathname]);

    if (shouldHideFooter) return null;

    return (
        <footer className="bg-dark" role="contentinfo">
            <div className="container mx-auto w-full py-[50px] px-10 md:px-0 space-y-12">
                <section className="grid gap-10 grid-cols-1 md:grid-cols-[3fr_2fr_2fr_1fr] text-white text-center md:text-start">
                    <div className="space-y-6 flex flex-col items-center md:items-start md:pr-40">
                        <LogoWhite aria-hidden="true" />
                        <p className="font-light text-sm">
                            DataRace is an innovative platform designed to bring data scientists and AI enthusiasts together to compete in data-driven challenges.
                        </p>
                    </div>
                    <div className="space-y-6">
                        <h2 className="font-semibold">Company</h2>
                        <ul className="text-sm font-light space-y-4">
                            <li>
                                <Link href="/about-us" className="hover:text-primary">About</Link>
                            </li>
                            <li>
                                <Link href="/faq" className="hover:text-primary">FAQ</Link>
                            </li>
                            <li>
                                <Link href="#" className="hover:text-primary">Terms and privacy policy</Link>
                            </li>
                        </ul>
                    </div>
                    <div className="space-y-6">
                        <h2 className="font-semibold">Races</h2>
                        <ul className="text-sm font-light space-y-4">
                            <li>
                                <Link href="/races" className="hover:text-primary">Active races</Link>
                            </li>
                            <li>
                                <Link href="/courses" className="hover:text-primary">Courses</Link>
                            </li>
                            <li>
                                <Link href="#" className="hover:text-primary">Proqram təminatı mühəndisliyi</Link>
                            </li>
                        </ul>
                    </div>
                    <div className="space-y-6">
                        <h2 className="font-semibold">Contact us</h2>
                        <ul className="text-sm font-light space-y-4">
                            <li>
                                <Link href="#" className="hover:text-primary">Email</Link>
                            </li>
                            <li>
                                <Link href="#" className="hover:text-primary">Phone</Link>
                            </li>
                            <li>
                                <Link href="#" className="hover:text-primary">Address</Link>
                            </li>
                        </ul>
                    </div>
                </section>

                <section className="flex flex-col items-center gap-10 md:flex-row md:justify-between">
                    <span className="text-gray-400">DataRace</span>
                    <div className="flex space-x-5 justify-center">
                        <a href="#" className="text-gray-400 hover:text-gray-900 dark:hover:text-white">
                            <InstagramIcon />
                            <span className="sr-only" aria-label='Instagram page'>Instagram page</span>
                        </a>
                        <a href="#" className="text-gray-400 hover:text-gray-900 dark:hover:text-white">
                            <TwitterIcon />
                            <span className="sr-only" aria-label='Twitter page'>Twitter page</span>
                        </a>
                        <a href="#" className="text-gray-400 hover:text-gray-900 dark:hover:text-white">
                            <YoutubeIcon />
                            <span className="sr-only" aria-label='YouTube page'>YouTube page</span>
                        </a>
                        <a href="#" className="text-gray-400 hover:text-gray-900 dark:hover:text-white">
                            <LinkedinIcon />
                            <span className="sr-only" aria-label='LinkedIn page'>LinkedIn page</span>
                        </a>
                    </div>
                </section>

            </div>
        </footer>
    );
};
