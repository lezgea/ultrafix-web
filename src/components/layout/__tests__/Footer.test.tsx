import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Footer } from '../footer';
import { usePathname } from 'next/navigation';

// Mock the dynamic imports
jest.mock('next/dynamic', () => () => (props: any) => <div>{props.children}</div>);

// Mock the usePathname hook
jest.mock('next/navigation', () => ({
    usePathname: jest.fn(),
}));

describe('Footer Component', () => {
    it('should render the footer when the pathname is not in hideHeaderRoutes', () => {
        (usePathname as jest.Mock).mockReturnValue('/some-route');

        const { asFragment } = render(<Footer />);

        expect(screen.getByRole('contentinfo')).toBeInTheDocument();
        expect(screen.getByText('DataRace')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should not render the footer when the pathname is in hideHeaderRoutes', () => {
        (usePathname as jest.Mock).mockReturnValue('/sign-in');

        const { asFragment } = render(<Footer />);

        expect(screen.queryByRole('contentinfo')).not.toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render the company links', () => {
        (usePathname as jest.Mock).mockReturnValue('/some-route');

        const { asFragment } = render(<Footer />);

        expect(screen.getByText('Company')).toBeInTheDocument();
        expect(screen.getByText('About')).toBeInTheDocument();
        expect(screen.getByText('FAQ')).toBeInTheDocument();
        expect(screen.getByText('Terms and privacy policy')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render the social media icons', () => {
        (usePathname as jest.Mock).mockReturnValue('/some-route');

        const { asFragment } = render(<Footer />);

        expect(screen.getByLabelText('Instagram page')).toBeInTheDocument();
        expect(screen.getByLabelText('Twitter page')).toBeInTheDocument();
        expect(screen.getByLabelText('YouTube page')).toBeInTheDocument();
        expect(screen.getByLabelText('LinkedIn page')).toBeInTheDocument();
        expect(asFragment).toMatchSnapshot();
    });
});