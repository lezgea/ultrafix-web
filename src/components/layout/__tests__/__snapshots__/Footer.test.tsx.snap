// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should not render the footer when the pathname is in hideHeaderRoutes 1`] = `<DocumentFragment />`;

exports[`Footer Component should render the company links 1`] = `
<DocumentFragment>
  <footer
    class="bg-dark"
    role="contentinfo"
  >
    <div
      class="container mx-auto w-full py-[50px] px-10 md:px-0 space-y-12"
    >
      <section
        class="grid gap-10 grid-cols-1 md:grid-cols-[3fr_2fr_2fr_1fr] text-white text-center md:text-start"
      >
        <div
          class="space-y-6 flex flex-col items-center md:items-start md:pr-40"
        >
          <div />
          <p
            class="font-light text-sm"
          >
            DataRace is an innovative platform designed to bring data scientists and AI enthusiasts together to compete in data-driven challenges.
          </p>
        </div>
        <div
          class="space-y-6"
        >
          <h2
            class="font-semibold"
          >
            Company
          </h2>
          <ul
            class="text-sm font-light space-y-4"
          >
            <li>
              <a
                class="hover:text-primary"
                href="/about-us"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="/faq"
              >
                FAQ
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Terms and privacy policy
              </a>
            </li>
          </ul>
        </div>
        <div
          class="space-y-6"
        >
          <h2
            class="font-semibold"
          >
            Races
          </h2>
          <ul
            class="text-sm font-light space-y-4"
          >
            <li>
              <a
                class="hover:text-primary"
                href="/races"
              >
                Active races
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="/courses"
              >
                Courses
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Proqram təminatı mühəndisliyi
              </a>
            </li>
          </ul>
        </div>
        <div
          class="space-y-6"
        >
          <h2
            class="font-semibold"
          >
            Contact us
          </h2>
          <ul
            class="text-sm font-light space-y-4"
          >
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Email
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Phone
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Address
              </a>
            </li>
          </ul>
        </div>
      </section>
      <section
        class="flex flex-col items-center gap-10 md:flex-row md:justify-between"
      >
        <span
          class="text-gray-400"
        >
          DataRace
        </span>
        <div
          class="flex space-x-5 justify-center"
        >
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="Instagram page"
              class="sr-only"
            >
              Instagram page
            </span>
          </a>
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="Twitter page"
              class="sr-only"
            >
              Twitter page
            </span>
          </a>
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="YouTube page"
              class="sr-only"
            >
              YouTube page
            </span>
          </a>
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="LinkedIn page"
              class="sr-only"
            >
              LinkedIn page
            </span>
          </a>
        </div>
      </section>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer Component should render the footer when the pathname is not in hideHeaderRoutes 1`] = `
<DocumentFragment>
  <footer
    class="bg-dark"
    role="contentinfo"
  >
    <div
      class="container mx-auto w-full py-[50px] px-10 md:px-0 space-y-12"
    >
      <section
        class="grid gap-10 grid-cols-1 md:grid-cols-[3fr_2fr_2fr_1fr] text-white text-center md:text-start"
      >
        <div
          class="space-y-6 flex flex-col items-center md:items-start md:pr-40"
        >
          <div />
          <p
            class="font-light text-sm"
          >
            DataRace is an innovative platform designed to bring data scientists and AI enthusiasts together to compete in data-driven challenges.
          </p>
        </div>
        <div
          class="space-y-6"
        >
          <h2
            class="font-semibold"
          >
            Company
          </h2>
          <ul
            class="text-sm font-light space-y-4"
          >
            <li>
              <a
                class="hover:text-primary"
                href="/about-us"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="/faq"
              >
                FAQ
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Terms and privacy policy
              </a>
            </li>
          </ul>
        </div>
        <div
          class="space-y-6"
        >
          <h2
            class="font-semibold"
          >
            Races
          </h2>
          <ul
            class="text-sm font-light space-y-4"
          >
            <li>
              <a
                class="hover:text-primary"
                href="/races"
              >
                Active races
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="/courses"
              >
                Courses
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Proqram təminatı mühəndisliyi
              </a>
            </li>
          </ul>
        </div>
        <div
          class="space-y-6"
        >
          <h2
            class="font-semibold"
          >
            Contact us
          </h2>
          <ul
            class="text-sm font-light space-y-4"
          >
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Email
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Phone
              </a>
            </li>
            <li>
              <a
                class="hover:text-primary"
                href="#"
              >
                Address
              </a>
            </li>
          </ul>
        </div>
      </section>
      <section
        class="flex flex-col items-center gap-10 md:flex-row md:justify-between"
      >
        <span
          class="text-gray-400"
        >
          DataRace
        </span>
        <div
          class="flex space-x-5 justify-center"
        >
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="Instagram page"
              class="sr-only"
            >
              Instagram page
            </span>
          </a>
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="Twitter page"
              class="sr-only"
            >
              Twitter page
            </span>
          </a>
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="YouTube page"
              class="sr-only"
            >
              YouTube page
            </span>
          </a>
          <a
            class="text-gray-400 hover:text-gray-900 dark:hover:text-white"
            href="#"
          >
            <div />
            <span
              aria-label="LinkedIn page"
              class="sr-only"
            >
              LinkedIn page
            </span>
          </a>
        </div>
      </section>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer Component should render the social media icons 1`] = `[Function]`;
